provider/oci/oci.go

```diff
@@ -173,6 +209,7 @@ func (p *OCIProvider) Records(ctx context.Context) ([]*endpoint.Endpoint, error)
 				ZoneNameOrId:  zone.Id,
 				Page:          page,
 				CompartmentId: &p.cfg.CompartmentID,
+				Scope:         dns.GetZoneRecordsScopeEnum(zone.Scope),
```

Comment:
This parameter is optional, so you could skip it.

Reply:
zone scope (private) is required in order to list private zone. I think it have to stay.